docs/sql-migration-guide-upgrade.md

```diff
@@ -23,6 +23,7 @@ license: |
 {:toc}
 
 ## Upgrading From Spark SQL 2.4 to 3.0
+  - In Spark version 2.4 and earlier, SQL queries such as `FROM test` or `FROM test UNION ALL FROM test` are supported by accident. In hive-style `FROM test SELECT *`, the `SELECT` clause is not negligible. Therefore we will treat these queries as invalid since Spark 3.0.
```

Comment:
Actually, this is bigger than the error message changing because the old existing customer queries will not work.
How about making this as a PR title? Something like `Disallow FROM-only SQL statement`.

How do you think about this,  @gatorsmile ?

Reply:
Yeah. I think we can highlight it in the PR title. Also, the original motivation of this PR is to resolve the long standing and confusing error messages issued by our parser.

Reply:
@yeshengm BTW, we should also mention that neither Hive nor Presto ever supported the syntax `FROM <table-name>`. That's part of the reason why we considered it as "supported by accident".